pkg/metrics/status.go

```diff
@@ -126,19 +126,23 @@ func (s *statusCollector) Collect(ch chan<- prometheus.Metric) {
 		return
 	}
 
-	// Nodes and endpoints healthStatusResponse
-	var (
-		unreachableNodes     int
-		unreachableEndpoints int
-	)
+	// Endpoints healthStatusResponse
+	var unreachableEndpoints int
 
 	for _, nodeStatus := range healthStatusResponse.Payload.Nodes {
+		unreachableVal := float64(0)
 		for _, addr := range healthClientPkg.GetAllHostAddresses(nodeStatus) {
 			if healthClientPkg.GetPathConnectivityStatusType(addr) == healthClientPkg.ConnStatusUnreachable {
-				unreachableNodes++
+				unreachableVal = 1.0
 				break
 			}
 		}
+		ch <- prometheus.MustNewConstMetric(
+			s.unreachableNodesDesc,
+			prometheus.GaugeValue,
+			unreachableVal,
+			nodeStatus.Name,
+		)
```

Comment:
Always emitting this metric will result in a very high cardinality. IMHO it would be better to only emit this metric if it has a value of `1`

Reply:
I've changed it to only emit the metric on failure, as I know for sure that we'd have to exclude this metric if it was emitted from all nodes for all nodes and I am strongly assuming this is true for almost everyone